src/librustc_parse/parser/diagnostics.rs

```diff
@@ -934,6 +934,25 @@ impl<'a> Parser<'a> {
             return self.expect(&token::Semi).map(drop);
         } else if !sm.is_multiline(self.prev_token.span.until(self.token.span)) {
             // The current token is in the same line as the prior token, not recoverable.
+        } else if sm.is_multiline(self.prev_token.span.until(self.token.span))
+            && [token::Comma, token::Colon].contains(&self.token.kind)
+            && &self.prev_token.kind == &token::CloseDelim(token::Paren)
+        {
+            // Likely typo: The current token is on a new line and is expected to be
+            // `.`, `;`, `?`, or an operator after a close delimiter token.
+            //
+            // let a = std::process::Command::new("echo")
+            //         .arg("1")
+            //         ,arg("2")
+            //         ^
+            // https://github.com/rust-lang/rust/issues/72253
+
+            let sp = self.token.span;
+            self.struct_span_err(sp, &msg)
+                .span_label(self.token.span, "unexpected token")
+                .span_help(sp, "change this to `.`, `;`, `?`, or an operator")
```

Comment:
For future reference, we have "structured suggestions" for these kind of things (called through `.span_suggestion` , `.multipart_suggestion` and `.span_suggestions`, among other available methods) which lets other tools show tooltips for the code changes.

Reply:
Ty!  Will track them down and read up.

Reply:
I am collecting this information and plan to add it to the rustc-dev-guide diagnostics documentation.